src/ApiPort/CommandLine/ApiPortConfiguration.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Framework.Configuration;
+using Microsoft.Framework.OptionsModel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+
+namespace ApiPort.CommandLine
+{
+    internal static class ApiPortConfiguration
+    {
+        internal static TOptions Parse<TOptions>(IEnumerable<string> args, IDictionary<string, string> switchMapping)
+            where TOptions : class, new()
+        {
+            var booleanSwitches = GetConfigMapCounts(typeof(TOptions), t => t == typeof(bool), switchMapping);
+            var arrays = GetConfigMapCounts(typeof(TOptions), t => (t.IsGenericType && t.GetGenericTypeDefinition() == typeof(List<>)) || t.IsArray, switchMapping);
+            var updatedArgs = TransformArguments(args, arrays, booleanSwitches).ToArray();
+
+            var cmd = new ConfigurationBuilder(Environment.CurrentDirectory)
+                 .AddCommandLine(updatedArgs, switchMapping)
+                 .Build();
+
+            var manager = new ConfigureFromConfigurationOptions<TOptions>(cmd);
+
+            var options = new TOptions();
+            manager.Action(options);
+
+            return options;
+        }
+
+        private static IEnumerable<ConfigMapCount> GetConfigMapCounts(Type type, Func<Type, bool> comparer, IDictionary<string, string> switchMapping)
+        {
+            var arrayNames = type.GetProperties(BindingFlags.Instance | BindingFlags.Public)
+                .Where(b => comparer(b.PropertyType))
+                .Select(prop => prop.Name)
+                .ToList();
```

Comment:
nit: remove .ToList() so the evaluation of arrayNames happens lazily (while it is iterating through the foreach loop) instead of blocking here.